pyne/mcnp.py

```diff
@@ -1990,210 +1987,235 @@ def _read_meshtal_head(self, f):
+        x_bounds = [float(x) for x in f.readline().split()[2:]]
+        y_bounds = [float(x) for x in f.readline().split()[2:]]
+        z_bounds = [float(x) for x in f.readline().split()[2:]]
         # "Energy bin boundaries" contain one more word than "X boundaries"
-        self.e_bounds = [float(x) for x in f.readline().split()[3:]]
+        e_bounds = [float(x) for x in f.readline().split()[3:]]
+        return x_bounds, y_bounds, z_bounds, e_bounds
 
-        self.dims = [0, 0, 0] + [len(self.x_bounds) - 1,
-                                 len(self.y_bounds) - 1,
-                                 len(self.z_bounds) - 1]
+    def read_column_order(self, f):
+        """
+        Create dictionary with table headings as keys and their column
+        location as values. Dictionary is the private attribute _column_idx.
+
+        Parameters
+        ----------
+        f : str or filestream
+            Filestream of the meshtal file.
+    
+        Returns
+        -------
+        column_idx : dict
+            Dictionary of the column index.
+        """
 
         # skip blank line between enery bin boundaries and table headings
         f.readline()
 
-    def _read_column_order(self, f):
-        """Create dictionary with table headings as keys and their column
-        location as values. Dictionary is the private attribute _column_idx.
-        """
         line = f.readline()
         column_names = line.replace('Rel ', 'Rel_').replace(
             'Rslt * ', 'Rslt_*_').strip().split()
-        self._column_idx = dict(zip(column_names, range(0, len(column_names))))
+        column_idx = dict(zip(column_names, range(0, len(column_names))))
+        return column_idx
 
-    def _create_mesh(self, f, mesh_has_mats):
-        """Instantiate a Mesh object and tag the PyMOAB core instance
-           with results and relative errors.
+    def read_tally_results_rel_error(self, f, num_e_groups, num_ves,
+            column_idx):
         """
+        Read meshtally results and reletive error data.
+    
+        Parameters
+        ----------
+        f : str or filestream
+            Filestream of the meshtal file.
+        num_e_groups: int
+            Number of energy groups.
+        num_ves: int
+            Number of volume elements.
+        column_idx : dict
+            Dictionary of the column index.
 
-        mats = () if mesh_has_mats is True else None
-        super(MeshTally, self).__init__(structured_coords=[self.x_bounds,
-                                                           self.y_bounds, self.z_bounds],
-                                        structured=True, mats=mats)
-
-        num_vol_elements = (len(self.x_bounds)-1) * (len(self.y_bounds)-1)\
-            * (len(self.z_bounds)-1)
-        num_e_groups = len(self.e_bounds)-1
+        Returns
+        -------
+        results : numpy array
+            Tally results data, shape=(num_ves, num_e_groups)
```

Comment:
I'd put these shape comments in the line above next to the type.

Reply:
Changed in https://github.com/pyne/pyne/pull/1199/commits/86db650ff475201d4cf61d1e8ea1c04015bdaf98.